source/grid/grid_tools.cc

```diff
@@ -5110,7 +5110,22 @@ namespace GridTools
           }
       }
     } // namespace DistributedComputePointLocations
-  }   // namespace internal
+
+
+    template <typename T, typename U>
+    std::vector<T>
+    permute(const std::vector<T> &input, const std::vector<U> &indices)
```

Comment:
That's the kind of function that would also have gone well into namespace `Utilities`.

If you wanted to do that, it would be useful to do this:
```suggestion
    permute(const std::vector<T> &input, const std::vector<typename std::vector<T>::size_type> &indices)
```
and drop the template type `U`.

Reply:
This is not possible since `U` is not `unsigned int` but a tuple `std::tuple<unsigned int, unsigned int, unsigned int>`. As stated above, this function will be gone after the refactoring.